lms/djangoapps/ccx/views.py

```diff
@@ -71,43 +71,60 @@ def wrapper(request, course_id):
         and modifying the view's call signature.
         """
         course_key = CourseKey.from_string(course_id)
+        ccx = None
+        if isinstance(course_key, CCXLocator):
+            # is there a security leak here in not checking that this user is
+            # the coach for this ccx?
+            ccx_id = course_key.ccx
+            ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
+            course_key = ccx.course_id
+
         role = CourseCcxCoachRole(course_key)
         if not role.has_user(request.user):
```

Comment:
@cpennington: I've added a check for this, but it happens *after* this code block here, so github is not picking it up as making this comment obsolete.